hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/HoodieSparkUtils.scala

```diff
@@ -103,7 +103,7 @@ object HoodieSparkUtils extends SparkAdapterSupport with SparkVersionsSupport {
         val transform: GenericRecord => GenericRecord =
           if (sameSchema) identity
           else {
-            HoodieAvroUtils.rewriteRecordDeep(_, readerAvroSchema)
+            HoodieAvroUtils.rewriteRecordDeep(_, readerAvroSchema, true)
```

Comment:
this is in the fast path and would like us to be cautious in doing any per record validations in general. We do have schema validations already. So, trying to wonder if this is really necessary? and why not rely on schema validations and not do per record validations.